keps/sig-storage/2848-transfer-volumesnapshots-namespaces/README.md

```diff
@@ -0,0 +1,801 @@
+- The target VolumeSnapshot will point to a newly created VolumeSnapshotContent.
+- Use Validation Webhook to reject any transfer contains secret.
+- Options to be introduced allowing transfer of VolumeSnapshot with secret for specific CSI Drivers (e.g. Ceph).
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Transfer of resources other than VolumeSnapshot will not be discussed.
+- Transfer of PersistentVolumeClaim may be discussed in the future. 
+- Transfer of VolumeSnapshots which contain secret by default.
+
+## Proposal
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+`StorageTransferRequest` resource:
+
+```yaml
+apiVersion: v1alpha1
+kind: StorageTransferRequest
+metadata:
+  name: transfer-foo
+  namespace: target-namespace
+spec:
+  source:
+    name: foo
+    namespace: source-namespace
+    kind: VolumeSnapshot
+  approvalName: approve-foo
+  targetName: bar
+```
+
+`StorageTransferApproval` resource:
+
+```yaml
+apiVersion: v1alpha1
+kind: StorageTransferApproval
+metadata:
+  name: approve-foo
+  namespace: source-namespace
+spec:
+  source:
+    name: foo
+    kind: VolumeSnapshot
+  targetNamespace: target-namespace
+  requestName: transfer-foo
+```
+
+When matching resources are detected, the controller initiates the transfer process.
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+Users should be able to transfer VolumeSnapshots across namespaces. This allows users to quickly share snapshots, utilizing data for testing
+or backups in separate namespaces. A new VolumeSnapshotContent must be created and bound to the new VolumeSnapshot, as the mapping between VolumeSnapshot and VolumeSnapshotContent is `1:1`.
+
+#### Story 1
+
+- The `prod` namespace contains a PVC that holds user information. VolumeSnapshots are taken of
+this PVC on a regular basis.
+
+- An existing VolumeSnapshot can be transferred to a test namespace. The user initiates the request
+by creating a `StorageTransferRequest` in the source namespace.
+
+- Once created, a `StorageTransferApproval` resource in the target namespace will be created.
+If matching resources are found, the VolumeSnapshot controller can init the transfer. 
```

Comment:
Isn't it the other way around?  Requests are in the source, Approvals in the source?

Reply:
@mhenriks initially I used Requests to be in the Target, however I changed following https://github.com/kubernetes/enhancements/pull/2849#discussion_r681990241